ftw/linkchecker/linkchecker.py

```diff
@@ -3,80 +3,58 @@
 import requests
 import time
 
-TIMEOUT = 1
-
 
 def millis():
     return int(round(time.time() * 1000))
 
 
-def get_uri_response(url_item):
+def get_uri_response(external_link_obj):
     headers = {
         'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36'
     }
     error = None
     response = None
     start_time = millis()
     try:
-        response = requests.head(url_item[2],
+        response = requests.head(external_link_obj.link_target,
                                  timeout=TIMEOUT,
                                  headers=headers,
-                                 allow_redirects=False)
+                                 allow_redirects=False,
+                                 verify=False)
     except requests.exceptions.Timeout:
         error = 'Timeout'
     except requests.exceptions.TooManyRedirects:
         error = 'Too many redirects'
     except requests.exceptions.ConnectionError:
         error = 'Connection Error'
-    except requests.exceptions.RequestException as e:
-        error = e
+    except Exception as e:
+        error = e.message
 
     time = millis() - start_time
 
-    attachment = extract_header_information(response)
-
-    if attachment['status code'] == 200 \
-            or 'resolveuid' in url_item[2]:
-        return []
+    if response and response.status_code == 200 \
+            or 'resolveuid' in external_link_obj.link_target:
+        external_link_obj.is_broken = False
     else:
-        return [
-            url_item[0],
-            url_item[1],
-            url_item[2],
-            attachment['status code'],
-            attachment['content type'],
-            time,
-            attachment['header location'],
-            error,
-        ]
+        external_link_obj.is_broken = True
+        external_link_obj.status_code = getattr(response, 'status_code', None)
+        external_link_obj.content_type = headers.get('Content-Type', None)
+        external_link_obj.response_time = time
+        external_link_obj.header_location = headers.get('Location', None)
+        external_link_obj.error_message = error
+
+    return external_link_obj
 
 
 def limit_cpu():
     os.nice(19)
 
 
-def extract_header_information(response):
-    attachment = {
-        'status code': None,
-        'header location': None,
-        'content type': None,
-    }
-    if not response:
-        return attachment
-
-    attachment['status code'] = response.status_code
-
-    headers = response.headers
-    if headers:
-        attachment['header location'] = headers.get('Location', None)
-        attachment['content type'] = headers.get('Content-Type', None)
-    return attachment
-
-
-def work_through_urls(urls):
+def work_through_urls(external_link_objs, timeout_config):
+    global TIMEOUT
```

Comment:
Instead of defining it like that you could just passt it to `get_uri_response`. 

setting a global variable in a function makes only an option if it's really necessary. 
Since it makes debugging and analysing the code unnecessary  more difficult. It could be set anywhere in that file

Reply:
I'm gonna do that in #44 so we can proceed with this pull.